ubiquity/dell-bootstrap.py

```diff
@@ -1023,7 +1100,25 @@ def build_rp(self, cushion=600):
                 if item.startswith('ID_FS_UUID'):
                     uuid = item.split('=')[1]
                     break
-
+        
+        #if select DHC, will copy DHC file for silent install
+        with misc.raised_privileges():
+            if self.rec_type=='dhc':
+                if os.path.exists(os.path.join("/mnt", "preseed", "dell-recovery.seed")):
+                    os.remove(os.path.join("/mnt", "preseed", "dell-recovery.seed"))
+                if not os.path.exists(os.path.join("/mnt", "scripts")):
+                    os.makedirs(os.path.join("/mnt", "scripts"))
+                dirs=["casper", "preseed", "scripts"]
+                for dir_tmp in dirs:
+                    mnt_dir=os.path.join("/mnt", "DHC", dir_tmp)
+                    assembly_dir=os.path.join("/mnt", dir_tmp)
+                    magic.white_tree("copy",re.compile(''), mnt_dir, assembly_dir)
+        #There is a script in platform_script that will copy the platform list file to the /usr/share/dell/DHC/ dir, so during recovery you can read the platform list. The way to do this is to  avoid recovery code change when add new platform, because you only need to update the file.     
```

Comment:
As in it's not owned by a package?  Then it shouldn't be put into `/usr/`.  It should be going into more likely something like `/var/lib/dell-recovery`

Reply:
It's owned by DHC package, when launching the dell recovery tool, need to read the platform ID file to determine whether to display the "recovery DHC" option. When just launched, I don’t know how to give root permission to mount the Recovery partition, so I copied this file to a directory during installation, and read this directory during recovery.  So you mean using the directory /var/lib/dell-recovery

Reply:
Yes, either a state file should be stored in the stateful directory (`/var/lib`) or the cache directory (`/var/cache`).  If you need it to persist across a reinstall, you can create something in EFI NVRAM (`/sys/firmware/efi/efivars`)